Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -8,54 +8,43 @@ public GildedRose(Item[] items) {
     }
 
     public void updateQuality() {
-        for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
-            } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
 
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
+        for (Item currItem : items) {
+            switch (currItem.getItemType()) {
+                case AGED_BRIE:
+                    currItem.checkQualityAndIncrease(1);
```

Comment:
consider moving that logic into seperate methods its a quite long logic block

Reply:
👍